### PR TITLE
Moved everything to sails logging

### LINKED ISSUE #428: Don't allow console.log and use sails logging instead

### PR DESCRIPTION
**Pull request by not-an-aardvark**
Fixes #428

### CODE REVIEW

File: api/controllers/UserController.js [hunk 1]
```diff
@@ -228,10 +222,9 @@ module.exports = {
         promises.push(Ban.localBanUser(req.params.username));
       }
       Promise.all(promises).then(function () {
-        console.log('Process to ban /u/' + req.params.username + ' was completed successfully.');
+        sails.log('Process to ban /u/' + req.params.username + ' was completed successfully.');
```
Comment (YaManicKill) [inline api/controllers/UserController.js]: does "sails.log" work without a level?

File: api/services/Ban.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ var _ = require('lodash');
 exports.banFromSub = async function (redToken, username, banMessage, banNote, subreddit, duration) {
   try {
     await Reddit.banUser(redToken, username, banMessage, banNote, subreddit, duration);
-    console.log('Banned ' + username + ' from /r/' + subreddit);
+    sails.log('Banned ' + username + ' from /r/' + subreddit);
```
Comment (YaManicKill) [inline api/services/Ban.js]: Again here

File: api/services/Ban.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ exports.giveBannedUserFlair = async function (redToken, username, current_css_cl
     var flair_text = current_flair_text || '';
     var css_class = Flairs.makeNewCSSClass(current_css_class, 'banned', subreddit);
     await Reddit.setUserFlair(redToken, username, css_class, flair_text, subreddit);
-    console.log('Changed ' + username + '\'s flair to ' + css_class + ' on /r/' + subreddit);
+    sails.log('Changed ' + username + '\'s flair to ' + css_class + ' on /r/' + subreddit);
```
Comment (YaManicKill) [inline api/services/Ban.js]: And here

File: api/services/Ban.js [hunk 4]
```diff
@@ -50,13 +49,13 @@ exports.updateAutomod = async function (redToken, username, subreddit, friend_co
...
   catch (automodparseerr) {
-    console.log('Error parsing /r/' + subreddit + ' AutoModerator config');
+    sails.log.error('Error parsing /r/' + subreddit + ' AutoModerator config');
     throw {error: 'Error parsing /r/' + subreddit + ' AutoModerator config'};
   }
   var content = lines.join("\n");
   await Reddit.editWikiPage(redToken, subreddit, 'config/automoderator', content, 'FlairHQ: Updated banned friend codes');
   var output = 'Added /u/' + username + '\'s friend codes to /r/' + subreddit + ' AutoModerator blacklist';
-  console.log(output);
+  sails.log(output);
```
Comment (YaManicKill) [inline api/services/Ban.js]: And here

File: api/services/Ban.js [hunk 5]
```diff
@@ -70,7 +69,7 @@ exports.markTSVThreads = async function (redToken, username) {
   });
   await Promise.all(tsv_promises);
   var output = 'Marked and locked /u/' + username + '\'s TSV threads (' + threads.length.toString() + ' total)';
-  console.log(output);
+  sails.log(output);
```
Comment (YaManicKill) [inline api/services/Ban.js]: Etc

File: api/controllers/UserController.js (hunk 1 above)
Comment (not-an-aardvark) [inline api/controllers/UserController.js]: Yes, `sails.log()` is an alias for `sails.log.debug()`. ([Docs](http://sailsjs.org/documentation/concepts/logging/sails-log))

**Pull request merged**